### PR TITLE
feat: add Chip component

### LINKED ISSUE #25: Migrate: DangerChip
- Introduced: https://github.com/reactioncommerce/reaction/pull/5200
- Used in this monster @kieckhafer WIP PR, https://github.com/reactioncommerce/reaction/pull/5158
- What is a chip!??!?! 

https://material-ui.com/components/chips

Chips: 

Screen Shot 2019-07-22 at 5 50 43 PM

### PR DESCRIPTION
**Pull request by cshepherd**
Resolves #25
Impact: **minor**  
Type: **feature**

## Component
Chip

## Testing
See docs for 'Chip'; Confirm that it is behaving as expected. Note that the variant 'outlined' is used as default, unlike regular MUI; This behavior was brought over directly from Reaction.

### CODE REVIEW

**Comment by machikoyasuda** [comment]
```10:45:51 AM: Building style guide...
10:46:13 AM:  FAIL  Failed to compile
10:46:13 AM: ./node_modules/react-styleguidist/lib/client/index.js (./node_modules/react-styleguidist/lib/loaders/styleguide-loader.js!./node_modules/react-styleguidist/lib/client/index.js)
10:46:13 AM: Error: Styleguidist: Section content file not found: /opt/build/repo/styleguide/src/sections/Content.md
10:46:13 AM:     at Array.map (<anonymous>)
10:46:13 AM:     at Array.map (<anonymous>)
10:46:13 AM:  @ ./node_modules/react-styleguidist/lib/client/index.js 37:19-71
10:46:13 AM:  @ multi @babel/polyfill ./styleguide/src/styles.css ./node_modules/react-styleguidist/lib/client/index
10:46:13 AM: error Command failed with exit code 1.```

Have to add a Content.md file. It can be blank.

**Comment by mikemurray** [comment]
probably just missing the `Content.md` file in "styleguide/src/sections/Content.md". 

One question I had is, are these section docs even useful? All we ever put in them is "This section is for components that do actions or whatever". With the collapsible menu I'm working on, it's also a little weird to have a document associated with the section header in the menu. When you click to expand/close, it also opens the document.

**Comment by machikoyasuda** [comment]
@mikemurray we can remove them i think. we're not using them/have nothing to put in there.

File: package/src/components/Chip/Chip.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+### Overview
+
+The Catalyst Chip inherits from the Material-UI [Chip component](https://material-ui.com/components/chips/). Refer to tne Material-UI [Chip API docs](https://material-ui.com/api/chip/) for more information.
```
Comment (machikoyasuda) [inline package/src/components/Chip/Chip.md]:
```suggestion
The Catalyst Chip inherits from the Material-UI [Chip component](https://material-ui.com/components/chips/). Refer to the Material-UI [Chip API docs](https://material-ui.com/api/chip/) for more information. 
```

Comment (machikoyasuda) [reply package/src/components/Chip/Chip.md]: Amusingly we had the same typo :D

**Comment by machikoyasuda** [comment]
Did something wonky happen with the merge commit from master? It looks like it pulled in a bunch of commits themselves instead.

**Comment by cshepherd** [comment]
will resubmit from new branch with cleaned-up history

**Pull request closed without merging**